sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -59,22 +59,27 @@ class RenderParagraph extends RenderBox {
     textPainter.minHeight = constraints.minHeight;
     textPainter.maxHeight = constraints.maxHeight;
     textPainter.layout();
+    // By default, we shrinkwrap to the intrinsic width.
+    double width = constraints.constrainWidth(textPainter.maxIntrinsicWidth);
```

Comment:
Oh my. So we're doing two layouts every time we want to get any metrics out of text? This seems unnecessarily expensive if all we're trying to do is position the text.

I don't understand the underlying problem. Is it that we're passing weird constraints in (e.g. the parent constraints unmodified), then shifting the text? Consider the case where the text is really long. Would it be going off to the left?